atlp-rwanda/ecommerce-app-legends-fe
# Chore esLint pre-commit hook #48

# Description
This PR sets up ESLint and a pre-commit hook to ensure that the codebase adheres to our coding style and passes all tests before every commit.

ESLint is now the required linter for the project, and it will check for style violations and flag them. The pre-commit hook has been configured to run two commands before every commit:

## Changes Made
Configured ESLint and Git pre-commit hook to run tests and ESLint format before commits. This ensures that style violations are flagged and that code is not committed with test breaking changes.

## Implementation Details
- Configured ESLint and Prettier to enforce code style guidelines
- Added pre-commit hook to format the code base before every commit
- Added test command to ensure no code is committed with test breaking changes

## How to Test
1. Clone this Repo
2. install dependancies `npm i`
3. Make changes to the code *by violating the linting style* ex: put extra lines beteen statements
5. Attempt to commit the changes without fixing any style violations or test breaking changes
6. Observe that the pre-commit hook prevents the commit and displays error messages

## Screenshots

<img width="743" alt="image" src="https://user-images.githubusercontent.com/46238001/233099839-d94e7d6a-f70b-4edd-851f-365cb27a9bff.png">


## Related Issues
None
